src/main/java/io/unlogged/core/bytecode/ClassTransformer.java

```diff
@@ -176,25 +214,130 @@ public void visitInnerClass(String name, String outerName,
      * Create an instance of a MethodVisitor that inserts logging code into a method.
      */
     @Override
-    public MethodVisitor visitMethod(int access, String name, String desc,
-                                     String signature, String[] exceptions) {
-        MethodVisitor mv = cv.visitMethod(access, name, desc, signature, exceptions);
-        if (name.equals("equals") || name.equals("hashCode")) {
-            return mv;
-        }
-        if (mv != null) {
-            mv = new TryCatchBlockSorter(mv, access, name, desc, signature, exceptions);
-            MethodTransformer trans = new MethodTransformer(
-                    weavingInfo, config, sourceFileName,
-                    fullClassName, outerClassName, access,
-                    name, desc, signature, exceptions, mv
-            );
-            return new JSRInliner(trans, access, name, desc, signature, exceptions);
-        } else {
-            return null;
+    public MethodVisitor visitMethod(int access, String name, String desc, String signature, String[] exceptions) {	
+		
+		// calcuclate probe flag at method level 
+		Boolean alwaysProbeMethodFlag = this.alwaysProbeClassFlag || ProbeFlagUtil.getalwaysProbeMethodFlag(name, access, desc);
+		Boolean neverProbeMethodFlag = ProbeFlagUtil.getneverProbeMethodFlag(name);
+
+		// early exit for clinit. It is already defined in class with initial method
+		if (name.equals("<clinit>")) {
+			this.hasStaticInitialiser = true;
+			MethodVisitor methodVisitorProbed = super.visitMethod(access, name, desc, signature, exceptions);
+
+			FieldVisitor fieldVisitor = visitField(Opcodes.ACC_STATIC, Constants.mapStoreCompileValue, "Ljava/util/HashMap;", null, null);
+			fieldVisitor.visitEnd();
+
+			methodVisitorProbed = new InitStaticTransformer(methodVisitorProbed, fullClassName, this.methodList);
+			methodVisitorProbed = addProbe(methodVisitorProbed, access, name, desc, exceptions);
+			return methodVisitorProbed;
+		}
+
+		// early exit got method that are never probed
+		if (neverProbeMethodFlag) {
+			MethodVisitor methodVisitorUnModified = cv.visitMethod(access, name, desc, signature, exceptions);
+            return methodVisitorUnModified;
         }
+		
+		// early exit for method that are always probed
+		if (alwaysProbeMethodFlag) {
+			MethodVisitor methodVisitorProbed = super.visitMethod(access, name, desc, signature, exceptions);
+			methodVisitorProbed = addProbe(methodVisitorProbed, access, name, desc, exceptions);
+			return methodVisitorProbed;
+		}
+
+		this.methodList.add(name);
+		String name_probed = name + "_PROBED";
```

Comment:
Can we use $$Unlogged$Probed as the suffix ?

Reply:
Yeah, I have changed method suffix to `$$Unlogged$Probed` now.